lightning/src/ln/chanmon_update_fail_tests.rs

```diff
@@ -926,7 +926,8 @@ fn do_test_monitor_update_fail_raa(test_ignore_second_cs: bool) {
 	expect_pending_htlcs_forwardable_and_htlc_handling_failed!(nodes[1], vec![HTLCDestination::NextHopChannel { node_id: Some(nodes[2].node.get_our_node_id()), channel_id: chan_2.2 }]);
 	check_added_monitors!(nodes[1], 1);
 
-	let mut events_3 = nodes[1].node.get_and_clear_pending_msg_events();
+	let mut events_3_vec = nodes[1].node.get_and_clear_pending_msg_events();
+	let events_3 = &mut events_3_vec;
```

Comment:
rather than making a mutable reference here and using two variables, could you explicitly insert the `&mut` prefix in the individual call sites?

Reply:
Unfortunately our rust 1.41.1 and 1.45.2 considers that as also taking a mutable reference borrow of the node itself and not only the list returned by `get_and_clear_pending_msg_events` and therefore fails the build. I found that super confusing as well :sweat_smile:. That was the CI failure that I got earlier in this PR. https://github.com/lightningdevkit/rust-lightning/pull/1966#issuecomment-1397777270

Reply:
hm, that is quite curious. But if that's how it is, I guess it'll have to be that way. In which case, could you add a comment and use slightly more descriptive variable names? Perhaps the first can be `events`, and the second one `mutable_events_reference` or something like that?

Reply:
Ah, that should work thanks! I was doing it the other way around which caused the issue for 1.41.1 and 1.45.2, but which worked for later builds, ie:
```
let events_3 = &mut nodes[1].node.get_and_clear_pending_msg_events();
let nodes_0_event = remove_first_msg_event_to_node(&nodes[0].node.get_our_node_id(), events_3);
```

My bad, sorry